lib/resourcebuilder/apps.go

```diff
@@ -72,6 +76,34 @@ func (b *deploymentBuilder) Do(ctx context.Context) error {
 		}
 	}
 
+	// the CVO is the only operator which doesn't have knowledge of its own operand (itself).  If we detect the CVO deployment
+	// we need to replace the KUBERNETES_SERVICE_HOST env var with the internal load balancer to be resilient to kube-apiserver
+	// rollouts that cause the localhost server to become non-responsive for multiple minutes.
+	if deployment.Namespace == "openshift-cluster-version" && deployment.Name == "cluster-version-operator" {
+		infrastructureConfig, err := b.infrastructureGetter.Infrastructures().Get("cluster", metav1.GetOptions{})
+		// not found just means that we don't have proxy configuration, so we should tolerate and fill in empty
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		lbURL, err := url.Parse(infrastructureConfig.Status.APIServerInternalURL)
```

Comment:
Sadly, API does not declare whether `APIServerInternalURL` is [optional or required][1].  But it _does_ declare [`Status` to be `+optional`][2].  So I think you need an `infrastructureConfig.Status == nil`, and possibly also an `infrastructureConfig.Status.APIServerInternalURL == ""`, guard which... falls back to `127.0.0.1` or `localhost` or some such.

[1]: https://github.com/openshift/api/blob/59994290fe0e13f63ae65f4c5a5f734b0d9f1eb4/config/v1/types_infrastructure.go#L74-L78
[2]: https://github.com/openshift/api/blob/59994290fe0e13f63ae65f4c5a5f734b0d9f1eb4/config/v1/types_infrastructure.go#L19-L21

Reply:
Ah, `Status` is `+optional`, but not a pointer (for mysterious, historical reasons).  No need for a `nil` guard.  Marking this thread resolved.